### DISCUSSION TITLE
Issues with Jest and `.env.test`

### DISCUSSION THREAD

**Discussion by cal-overflow** [goatandsheep/react-native-dotenv#464] [Q&A]
I have read both the docs and existing posts several times now. I currently have an issue when running my jest tests. \
Here's a minimal recreation of my setup:

`.env`
```
MY_VAR=real_value
```

`babel.config.js`
```js
module.exports = {
  presets: ['module:metro-react-native-babel-preset'],
  sourceMaps: true,
  plugins: ['module:react-native-dotenv'],
};
```

When I run my application, everything works as expected. The values from `.env` are loaded in correctly. Here's an example of how I use these variables in my project.
`src/WelcomeScreen.js`
```js
import { MY_VAR } from '@env';

console.log(`variable value: ${MY_VAR}`);
// ...
```

Like I mentioned above, this works when I build the app using metro. I would see `variable value: real_value` as the log output.

### The issue
The problem comes when I run [jest](https://jestjs.io/). Following the docs, I have created a special `.env.test` file with a different set of values.
`.env.test`
```
MY_VAR=test_value
```

When I run my test files, I see that these `.env.test` variable values are used. For example:
`src/WelcomeScreen.test.js`
```js
import { MY_VAR } from '@env';

console.log(`variable value: ${MY_VAR}`);
```
Shows `variable value: test_value`. \
However, the reference to `MY_VAR` in `src/WelcomeScreen.js` will throw the following error.
```
ReferenceError: MY_VAR is not defined.
```

At this point I'm unsure if there is a configuration setting I'm missing or there is something going on with Jest that is interfering with how the module is being loaded into the non-test files. \
Any suggestions are welcome!

Related:
- #59 
- #204

**Comment by cal-overflow** [comment]
@goatandsheep do you have any idea on how to resolve this?

**Comment by cal-overflow** [comment]
Still haven't found a solution for this. I just created a wrapper module that I can easily mock. It would be nice if there was clarity around this, though.

**Reply by goatandsheep** [reply]
I'm looking into it 👍🏽 I'd love to see what you did for the wrapper!

**Reply by goatandsheep** [reply]
@cal-overflow have you tried doing `jest --no-cache`?
